pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingDataTest.java

```diff
@@ -180,4 +183,32 @@ void limitNotChanged() {
         }
     }
 
+    @Test
+    void zeroLimitReadEmptyFile() throws IOException {
+        final Path file = Files.createTempFile(getClass().getSimpleName(), "zeroLimitReadEmptyFile");
+        try (final var stream = new ReadableStreamingData(file)) {
+            assertThat(stream.limit()).isEqualTo(0);
+        }
+    }
+
+    @Test
+    void nonZeroLimitReadFile() throws IOException {
+        final Path file = Files.createTempFile(getClass().getSimpleName(), "nonZeroLimitReadFile");
```

Comment:
Should the test delete the file? The spec for Files.createTempFile() says that one needs to use `DELETE_ON_CLOSE`, or call `File.deleteOnExit()` for this to actually happen.

Reply:
Good catch. Added code to delete the temp files on exit.